src/modules/user/core/services/kyc.service.new.ts

```
@@ -0,0 +1,175 @@
+    const userIdEntity = EntityId.fromString(uid);
+    const currentKyc = await this.kycRepository.getByUserId(userIdEntity);
+
+    if (!currentKyc) {
+      throw new NotFoundError(`KYC_DATA for ${uid}`);
+    }
+
+    const kycResourceData = await this.retrieveKycData(resourceUrl);
+
+    const { country, subtype, type, fields } = kycResourceData.documents[0];
+    const documentNumber = fields.documentNumber.value;
+    const curp = this.constants.METAMAP_SUBTYPES.MX.includes(subtype) ? fields.curp.value : null;
+    const address = this.constants.METAMAP_SUBTYPES.MX.includes(subtype) ? fields.address.value : null;
+
+    currentKyc.country = country;
+    currentKyc.address = address;
+    currentKyc.curp = curp;
+    currentKyc.subtype = subtype ?? type;
+    currentKyc.documentNumber = documentNumber;
+    if (await this.kycRepository.update(currentKyc)) {
+      return { success: true, message: 'KYC data updated successfully' };
+    } else {
+      return { success: false, message: 'KYC data update failed' };
+    }
+  }
+
+  async updateUserVerified(uid: string, status: MetamapStatus): Promise<{ success: boolean; message: string }> {
+    const userIdEntity = EntityId.fromString(uid);
+    const user = await this.userRepository.getById(userIdEntity);
+
+    user.kycStatus = status;
+    if (await this.userRepository.update(user)) return { success: true, message: 'KYC status updated successfully' };
+    return { success: false, message: 'KYC status update failed' };
+    // TODO: notification service
+  }
+
+  async cancelKyc(uid: string): Promise<{ success: boolean; message: string }> {
+    const userIdEntity = EntityId.fromString(uid);
+    const user = await this.userRepository.getById(userIdEntity);
+
+    if (user.kycStatus === MetamapStatus.VERIFIED) {
+      throw new Error('Cannot cancel KYC status for an already verified user.');
+    }
+
+    if (user.kycStatus !== 'VERIFIED') {
```

I see hardcoded strings here, also, if the user is already verified you should throw an error instead